src/main/scala/com/gu/riffraff/artifact/BuildInfo.scala

```diff
@@ -70,7 +70,7 @@ object BuildInfo {
       tcProps <- loadProps(tcPropFile)
       buildIdentifier <- prop("build.number", tcProps)
       revision <- prop("build.vcs.number", tcProps)
-      branch <- prop("teamcity.build.branch", tcProps)
+      branch <- prop("vcsroot.branch", tcProps).map(ref => ref.split("/").lastOption.getOrElse(ref))
```

Comment:
Maybe do a filter(_.nonEmpty) on the property value and then try falling back?
 Split on an empty string would result in an empty string.
```
scala> "".split("/").lastOption
res1: Option[String] = Some()
```